tests/RedshiftDatatypeTest.php

```diff
@@ -24,113 +24,33 @@ public function testInvalidType()
+     * @param string $expectedOutput
+     * @throws InvalidOptionException
+     * @dataProvider validLengthsProvider
      */
-    public function testInvalidNumericLengths($length)
+    public function testValidLengths($columnType, $option, $expectedOutput)
     {
-        try {
-            new Redshift("NUMERIC", ["length" => $length]);
-            $this->fail("Exception not caught");
-        } catch (\Exception $e) {
-            $this->assertEquals(InvalidLengthException::class, get_class($e));
-        }
-    }
+        $redshift = new Redshift($columnType, $option);
 
-    public function testValidVarcharLengths()
-    {
-        new Redshift("varchar");
-        new Redshift("VARCHAR");
-        new Redshift("VARCHAR", ["length" => ""]);
-        new Redshift("VARCHAR", ["length" => "1"]);
-        new Redshift("VARCHAR", ["length" => "65535"]);
-        new Redshift("VARCHAR", ["length" => ['character_maximum' => 65535]]);
-        new Redshift("VARCHAR", ["length" => []]);
+        $this->assertEquals($expectedOutput, $redshift->getLength());
     }
 
     /**
-     * @dataProvider invalidVarcharLengths
-     * @param $length
+     * @param string $columnType
+     * @param array $options
+     * @dataProvider invalidLengthsProvider
      */
-    public function testInvalidVarcharLengths($length)
+    public function testInvalidLengths($columnType, $options)
     {
-        try {
-            new Redshift("VARCHAR", ["length" => $length]);
-            $this->fail("Exception not caught");
-        } catch (\Exception $e) {
-            $this->assertEquals(InvalidLengthException::class, get_class($e));
-        }
-    }
-
-    /**
-     * @dataProvider timestampLengths
-     * @param $type
-     * @param $length
-     * @param $expectedValid
-     */
-    public function testTimestampLengths($type, $length, $expectedValid)
-    {
-        if (!$expectedValid) {
-            $this->expectException(InvalidLengthException::class);
-            new Redshift($type, ["length" => $length]);
-        } else {
-            new Redshift($type, ["length" => $length]);
-        }
-    }
-
-    public function testValidCharLengths()
-    {
-        new Redshift("char");
-        new Redshift("CHAR");
-        new Redshift("CHAR", ["length" => "1"]);
-        new Redshift("CHAR", ["length" => "4096"]);
-    }
-
-    /**
-     * @dataProvider invalidCharLengths
-     * @param $length
-     */
-    public function testInvalidCharLengths($length)
-    {
-        try {
-            new Redshift("CHAR", ["length" => $length]);
-            $this->fail("Exception not caught");
-        } catch (\Exception $e) {
-            $this->assertEquals(InvalidLengthException::class, get_class($e));
+        foreach ($options as $option) {
+            try {
+                new Redshift($columnType, ['length' => $option]);
+            } catch (\Exception $e) {
+                $this->assertEquals(InvalidLengthException::class, get_class($e));
```

Comment:
expect exception class + expect exception message?

Reply:
Ale chápu to jako minimální diff

Reply:
když jsem používal expectException tak to neprojede celý foreach

![image](https://user-images.githubusercontent.com/12143866/78012540-64d77180-7345-11ea-9e02-b54be91274d3.png)

první je s ExceptException
druhý tak jak to mám teď

Reply:
Jasně!